### PR TITLE
Add a check for the <script> tag

### PR DESCRIPTION
**Pull request by carolinan**
Fixes https://github.com/WordPress/theme-check/issues/149

It does not stop upload since there can be valid exceptions that needs manual review.

### CODE REVIEW

File: checks/script_tag.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+		$ret = true;
+
+		checkcount();
+		/**
+		 * This check is limited to header.php and footer.php.
+		 */
+		foreach ( $php_files as $file_path => $file_content ) {
+			$filename = tc_filename( $file_path );
+			if ( 'footer.php' === $filename && preg_match( '/<script/', $file_content ) || 'header.php' === $filename && preg_match( '/<script/', $file_content )) {
```
Comment (dd32) [inline checks/script_tag.php:19]:
```suggestion
			if ( ! in_array( $filename, array( 'header.php', 'footer.php' ) ) ) {
				continue;
			}

			if ( preg_match( '/<script/i', $file_content ) ) {
```

Seems much easier to read in this case, and checking for `<script` case insensitively.

File: checks/script_tag.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+				$error         = '/<script/';
+				$grep          = tc_preg( $error, $file_path );
+				$this->error[] = sprintf( '<span class="tc-lead tc-required">' . __( 'REQUIRED', 'theme-check' ) . '</span>: ' . __( 'Found a script tag in %s. Scripts and styles needs to be enqueued or added via a hook, not hard coded.', 'theme-check' ),
```
Comment (dd32) [inline checks/script_tag.php:22]: A common thing I'm seeing in Theme Check is that there's often a lack of spaces between error and the examples. This seems simpler and easier to read:

```suggestion
				$this->error[] = sprintf(
					'<span class="tc-lead tc-required">' . __( 'REQUIRED', 'theme-check' ) . '</span>: %s %s',
					sprintf(
						__( 'Found a script tag in %s. Scripts and styles needs to be enqueued or added via a hook, not hard coded.', 'theme-check' ),
						'<strong>' . $filename . '</strong>'
					),
					$grep
				);
```

File: checks/script_tag.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+				'<strong>' . $filename . '</strong>' ) . $grep;
```
Comment (dd32) [inline checks/script_tag.php:23]:
```suggestion
```

**Review by dd32 — CHANGES_REQUESTED** [review]
Just some thoughts on making the check read a bit simpler, and error formatting.

File: checks/script_tag.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+		 * This check is limited to header.php and footer.php.
+		 */
+		foreach ( $php_files as $file_path => $file_content ) {
+			$filename = tc_filename( $file_path );
+			if ( ! in_array( $filename, array( 'header.php', 'footer.php' ) ) ) {
+				continue;
+			}
+
+			if ( preg_match( '/<script/i', $file_content ) ) {
+				$error         = '/<script/';
```
Comment (carolinan) [inline checks/script_tag.php:24]:
```suggestion
				$error         = '/<script/i';
```

**Pull request merged**